mycodo/controllers/controller_pid.py

```diff
@@ -327,36 +361,29 @@ def setup_method(self, method_id):
         if self.method_type == 'Duration':
             if self.method_start_act == 'Ended':
                 # Method has ended and hasn't been instructed to begin again
-                pass
+                with session_scope(MYCODO_DB_PATH) as db_session:
+                    mod_pid = db_session.query(PID)
+                    mod_pid = mod_pid.filter(
+                        PID.unique_id == self.unique_id).first()
+                    mod_pid.is_activated = False
+                    db_session.commit()
+                self.stop_controller()
+                self.logger.warning(
+                    "Method has ended. "
+                    "Activate the PID controller to start it again.")
             elif self.method_start_act == 'Ready' or self.method_start_act is None:
                 # Method has been instructed to begin
                 now = datetime.datetime.now()
                 self.method_start_time = now
-                if method_data_repeat and method_data_repeat.duration_end:
+                if method.method_data_repeat and method.method_data_repeat.duration_end:
                     self.method_end_time = now + datetime.timedelta(
-                        seconds=float(method_data_repeat.duration_end))
+                        seconds=float(method.method_data_repeat.duration_end))
 
                 with session_scope(MYCODO_DB_PATH) as db_session:
                     mod_pid = db_session.query(PID).filter(PID.unique_id == self.unique_id).first()
                     mod_pid.method_start_time = self.method_start_time
                     mod_pid.method_end_time = self.method_end_time
                     db_session.commit()
-            else:
-                # Method neither instructed to begin or not to
-                # Likely there was a daemon restart ot power failure
-                # Resume method with saved start_time
-                self.method_start_time = datetime.datetime.strptime(
-                    str(pid.method_start_time), '%Y-%m-%d %H:%M:%S.%f')
-                if method_data_repeat and method_data_repeat.duration_end:
-                    self.method_end_time = datetime.datetime.strptime(
-                        str(pid.method_end_time), '%Y-%m-%d %H:%M:%S.%f')
-                    if self.method_end_time > datetime.datetime.now():
-                        self.logger.warning("Resuming method {id}: started {start}, ends {end}".format(
-                            id=method_id, start=self.method_start_time, end=self.method_end_time))
-                    else:
-                        self.method_start_act = 'Ended'
-                else:
-                    self.method_start_act = 'Ended'
```

Comment:
I'm not a 100% sure on this. But this never existed in the trigger controller ant tests have shown that it works without it in this new implementation.